source/javascripts/controllers/_MainController.js.erb

```diff
@@ -927,6 +923,68 @@ import {segmentTrack} from "@/utils/segmentTracking";
         });
       };
 
+      viewModel.saveSecretsFromReact = function (secretsFromReact) {
+        if (!appService.secrets) {
+          appService.secrets = [];
+        }
+
+        // delete the ones that got removed.
+        appService.secrets = appService.secrets.filter((secret) => {
+          // leave workspace secrets intact
+          if (secret.scope() === 'workspace') return true;
+
+          // remove the ones that are not in the react list
+          return secretsFromReact.filter(s => !s.isShared).find((s) => s.key === secret.key())
+        });
+
+        // edit existing ones
+        appService.secrets.forEach(secret => {
+          if (secret.scope() === 'workspace') return;
+
+          const s = secretsFromReact.find((s) => s.key === secret.key() && !s.isShared);
+
+          if (s) {
+            secret.value(s.value);
+            secret.isProtected(s.isProtected);
+            secret.isExpose(s.isExpose);
+            secret.isExpand(s.isExpand);
+            secret.isKeyChangeable = s.isKeyChangeable;
+          }
+
+          return secret;
+        });
+
+        // add new ones
+        secretsFromReact.forEach(secret => {
+          if (secret.isShared) return;
+
+          const existingSecret = appService.secrets
+            .filter(s => s.scope() !== 'workspace')
```

Comment:
If I understand correctly `isShared` and `scope() == 'workspace'` is the same, right? We should use the named function in that case everywhere.

Reply:
Technically yes, but the Angular secret and React secret objects differ a bit, and the Angular one only has the `scope(newScope)` getter/setter function.